packages/kit/src/module/define.ts

```diff
@@ -67,7 +67,25 @@ export function defineNuxtModule<OptionsT extends ModuleOptions> (definition: Mo
     }
 
     // Call setup
-    await definition.setup?.call(null as any, _options, nuxt)
+    const setupStart = process.hrtime()
+    const res = await definition.setup?.call(null as any, _options, nuxt) ?? {}
+    const setupEnd = process.hrtime(setupStart)
+
+    // Measure setup time
+    const setupTime = Math.round((setupEnd[0] * 1000000000 + setupEnd[1]) / 10000) / 100 /* round by two digits */
+    if (setupTime > 1000) {
+      logger.warn(`Module \`${uniqueKey || '<no name>'}\` took \`${setupTime}ms\` to setup`)
+    }
```

Comment:
* What about printing time of each module (however short) if `nuxt.options.debug` is set?

* Playing devil's advocate, we might also consider suppressing this warning when we first release the feature until we have a chance to test with the bigger ecosystem. In patch release, we could print warning only if module is _not_ located in `node_modules`, which might give authors a chance to rewrite. (Counter-point: all modules taking this long _should_ rewrite and we don't want users thinking _nuxt_ is slow.)

Reply:
I like the idea of displaying the timing for each module when having `DEBUG=1`, we could highlight the time I guess or add a `(slow)` next to it.

Reply:
I also prefer not to log them by default，it could be useful to module authors but not too much for regular users, but just make the output verbose.

In devtools, we could present them in UI with different levels of colors (yellow, orange, red) and be able to sort them - this way, authors could find out the bottleneck easier.

Reply:
Thanks for the feedback. I think we can show a little bit more verbose output for debug mode too (less threshold than 1000). Most modules are initialized less than few/1ms.

Reply:
Moved threshold to 5 seconds. It is really unlikely (and undesired for me) a module alone takes this long for setup. We could probably show much better inspection from devtools from meta anyway.